tools/integration_tests_runner.py

```diff
@@ -1,137 +1,162 @@
+            params[DATA_PATH_KEY] = tmpdir
 
-        if (not start and not finish) or (start and finish):
-            logging.warning("You need to pass either start=True or finish=True, but only one of them! You passed start={0}, finish={1}".format(start, finish))
-            return
+        keys = params_from_dict(params)
+        out, err, result = execute_file(test_file, keys)
 
-        string = "Started" if start else "Finished"
 
-        with self.start_finish_lock:
-            self.start_finish_log.write("{string} {test_file} {keys} at {time}\n".format(string=string, test_file=test_file, keys=keys, time=time()))
-            self.start_finish_log.flush()
+    except:
+        for tup in traceback.extract_stack():
+            multiprocessing.get_logger().error(tup)
 
+    finally:
+        if test_file in TEMPFOLDER_TESTS:
+            shutil.rmtree(tmpdir)
 
-    def exec_test(self, test_file, test, clean_env=False):
-        keys = '"--filter={test}"'.format(test=test)
-        if clean_env:
-            tmpdir = tempfile.mkdtemp()
-            keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
-            logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
-        else:
-            keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=self.key_postfix)
-            logging.debug("Setting user_resource_path and data_path to {resource_path}".format(resource_path=self.key_postfix))
+    return (test_file, test_name), (err, result)
 
-        self.log_start_finish(test_file, keys, start=True)
-        out, err, result = self.get_tests_from_exec_file(test_file, keys)
-        self.log_start_finish(test_file, keys, finish=True)
 
-        if clean_env:
-            try:
-                shutil.rmtree(tmpdir)
-            except:
-                logging.error("Failed to remove tempdir {tmpdir}".format(tmpdir=tmpdir))
+def params_from_dict(params_dict):
+    return " ".join( '"{0}={1}"'.format(key, value)
+        for key, value in params_dict.iteritems()
+        if value
+    )
 
-        with self.file_lock:
-            self.file.write("BEGIN: {file}\n".format(file=test_file))
-            self.file.write(str(err))
-            self.file.write("\nEND: {file} | result: {res}\n\n".format(file=test_file, res=result))
-            self.file.flush()
 
+class IntegrationRunner:
+    def __init__(self):
+        proc_count = cpu_count() / 5 * 4
+        self.pool = multiprocessing.Pool(proc_count, initializer=setup_main_logger)
+        self.params = {}
+        self.process_cli()
+        multiprocessing.get_logger().info(
+            "\n{0}\nIntegration tests runner started.\nMy pid is: {1}\n{0}\n".format(
+                "*" * 80,
+                getpid()
+            )
+        )
+
+
+    def run_tests(self):
+        logger = logging.getLogger(TEST_RESULT_LOG)
+        try:
+            for (test_file, test_name), (err, result) \
+                    in self.pool.imap_unordered(
```

Comment:
Pls, do it without \. \ - is for exceptional cases. Create var for self.pool.imap_unordered(exec_test, self.prepare_list_of_tests()) instead